src/components/home/BusinessesForm.js

```diff
@@ -11,13 +11,130 @@ class BusinessesForm extends Component {
       searchText: '',
       showDropdown: false,
       showPreviewDropdown: false,
+      actualSelectedItem: null,
+      currentIndex: null,
+      keyPress: false,
     };
+    this.defaultList = null;
+    this.resultsListHTML = null;
+    this._formOnKeyDown = this._formOnKeyDown.bind(this);
+    this._onMouseOver = this._onMouseOver.bind(this);
+    this._fromNodeListToArray = this._fromNodeListToArray.bind(this);
+    this._move = this._move.bind(this);
+    this.noop = this.noop.bind(this);
+    this._onMouseMove = this._onMouseMove.bind(this);
   }
+
+  noop() {}
+
+  _onMouseMove() {
+    this.setState({keyPress: false});
+  }
+
+  _formOnKeyDown(event) {
+    switch (event.key) {
+      case 'Enter':
+        event.preventDefault();
+        if (this.state.actualSelectedItem) {
+          this.state.actualSelectedItem.firstChild.click();
+        }
+        break;
+      case 'ArrowUp':
+        this._move('up');
+        break;
+      case 'ArrowDown':
+        this._move('down');
+        break;
+      default:
+        break;
+    }
+  }
+
+  _fromNodeListToArray(nodeList) {
+    this.resultsListHTML = nodeList;
+    this.defaultList = nodeList && [...nodeList.childNodes];
+  }
+
+  _move(action) {
+    this._fromNodeListToArray(this.defaultList);
+    const index = this.state.currentIndex;
+    const elem = this.state.actualSelectedItem;
+    let newIndex = null;
+    let newElem = null;
+
+    switch (action) {
+      case 'up':
+        if (index != null && elem != null) {
+          newIndex = index - 1;
+          if (newIndex < 0) {
+            newIndex = this.defaultList.length - 1;
+          }
+          newElem = this.defaultList[newIndex];
+        } else {
+          newIndex = this.defaultList.length - 1;
+          newElem = last(this.defaultList);
+        }
+        break;
+      case 'down':
+        if (index != null && elem != null) {
+          newIndex = index + 1;
+          if (newIndex > this.defaultList.length - 1) {
+            newIndex = 0;
+          }
+          newElem = this.defaultList[newIndex];
+        } else {
+          newIndex = 0;
+          newElem = this.defaultList[newIndex];
+        }
+        break;
+      default:
+        break;
+    }
+
+    if (newElem) {
+      this.resultsListHTML.scrollTo(0, newElem.offsetTop);
+    }
+
+    this.setState({
+      actualSelectedItem: newElem,
+      currentIndex: newIndex,
+      keyPress: true,
+    });
+  }
+
+  _onMouseOver(e) {
+    let elemLi;
```

Comment:
`const elemLi = e.target.tagName.toLowerCase() === 'a' ? e.target.parentElement : e.target;`

Lines 108 to 112 can be replaced with that ternary operator.